src/Microsoft.PowerShell.Commands.Utility/commands/utility/WaitEventCommand.cs

```diff
@@ -84,13 +102,13 @@ protected override void ProcessRecord()
 
             while (!received)
             {
-                if (_timeoutInSeconds >= 0)
+                if (_timeoutTimespan.TotalMilliseconds >= 0)
                 {
-                    if ((DateTime.UtcNow - startTime).TotalSeconds > _timeoutInSeconds)
+                    if ((DateTime.UtcNow - startTime) > _timeoutTimespan)
                         break;
                 }
 
-                received = _eventArrived.WaitOne(200);
+                received = _eventArrived.WaitOne((int)(_timeoutTimespan.TotalMilliseconds / 100));
```

Comment:
What is minimal value the WaitOne accept?

Reply:
@iSazonov thanks for asking.  I'm going to use an absolute value so that it always remains positive.